vortex-array/src/arcref.rs

```diff
@@ -72,16 +73,27 @@ where
     }
 }
 
+impl<T> Eq for ArcRef<T> where T: ?Sized + 'static + Eq {}
+
 impl<S, T> PartialOrd<ArcRef<S>> for ArcRef<T>
 where
     S: ?Sized + 'static,
     T: ?Sized + 'static + PartialOrd<S>,
 {
-    fn partial_cmp(&self, other: &ArcRef<S>) -> Option<std::cmp::Ordering> {
+    fn partial_cmp(&self, other: &ArcRef<S>) -> Option<Ordering> {
         self.deref().partial_cmp(other.deref())
     }
 }
 
+impl<T> Ord for ArcRef<T>
+where
+    T: ?Sized + 'static + Ord,
```

Comment:
out of curiosity why does T need to be 'static here? Is it beause how deref is implemented?